### PR TITLE
Manage Teams identity quickstart

### PR DESCRIPTION
**Pull request by AikoBB**
## Purpose
Quickstart for setting up and managing access tokens for Teams users

## Does this introduce a breaking change?

```
[ ] Yes
[x] No
```

## Pull Request Type
What kind of change does this Pull Request introduce?

- added quickstart for managing access token for Teams user

```
[ ] Bugfix
[x] Feature
[ ] Code style update (formatting, local variables)
[ ] Refactoring (no functional changes, no api changes)
[ ] Documentation content changes
[ ] Other... Please describe:
```

## How to Test
*  Get the code

```
git clone [repo-address]
cd [repo-name]
Open `ManageTeamsIdentityMobileAndDesktop.csproj`
Run the `ManageTeamsIdentityMobileAndDesktop` project
```

* Test the code

```
```

## What to Check
Verify that the following are valid

- You should be navigated from your browser to a standard OAuth flow with MSAL
- After successful authentication, the Azure AD access token should be exchanged for a Communication access token

### CODE REVIEW

File: ManageTeamsIdentityMobileAndDesktop/ManageTeamsIdentityMobileAndDesktop.csproj [hunk 1]
```diff
@@ -0,0 +1,15 @@
+<Project Sdk="Microsoft.NET.Sdk">
+
+  <PropertyGroup>
+    <OutputType>Exe</OutputType>
+    <TargetFramework>net6.0</TargetFramework>
+    <ImplicitUsings>enable</ImplicitUsings>
+    <Nullable>enable</Nullable>
```
Comment (petrsvihlik) [inline ManageTeamsIdentityMobileAndDesktop/ManageTeamsIdentityMobileAndDesktop.csproj:7]: consider changing this to `disable` or fixing the warnings generated by `Environment.GetEnvironmentVariable`

File: ManageTeamsIdentityMobileAndDesktop/Program.cs [hunk 2]
```diff
@@ -0,0 +1,52 @@
...
+            string teamsUserAadToken = result.AccessToken;
+            string userObjectId = result.UniqueId;
+
+            // This code demonstrates how to fetch your connection string
+            // from an environment variable.
+            string connectionString = Environment.GetEnvironmentVariable("COMMUNICATION_SERVICES_CONNECTION_STRING");
+            // Instantiate the identity client
+            var client = new CommunicationIdentityClient(connectionString);
+
+            //Exchange the Azure AD access token of the Teams User for a Communication Identity access token
```
Comment (petrsvihlik) [inline ManageTeamsIdentityMobileAndDesktop/Program.cs:46]:
```suggestion
            // Exchange the Azure AD access token of the Teams User for a Communication Identity access token
```

File: ManageTeamsIdentityMobileAndDesktop/Program.cs [hunk 3]
```diff
@@ -0,0 +1,52 @@
...
+            string redirectUri = "http://localhost";
+
+            // Create an instance of PublicClientApplication
+            var aadClient = PublicClientApplicationBuilder
+                            .Create(appId)
+                            .WithAuthority(authority)
+                            .WithRedirectUri(redirectUri)
+                            .Build();
+
+            List<string> scopes = new List<string> {
```
Comment (petrsvihlik) [inline ManageTeamsIdentityMobileAndDesktop/Program.cs:28]:
```suggestion
            List<string> scopes = new() {
```

File: ManageTeamsIdentityMobileAndDesktop/README.md [hunk 4]
```diff
@@ -0,0 +1,50 @@
...
+languages:
+- csharp
+products:
+- azure
+- azure-communication-services
+---
+
+# Create and manage Communication access tokens for Teams users in mobile and desktop applications
+
+This code sample walks you through the process of acquiring a Communication Token Credential by exchanging an Azure AD token of a user with a Teams license for a valid Communication access token.
```
Comment (petrsvihlik) [inline ManageTeamsIdentityMobileAndDesktop/README.md:12]: > Communication Token Credential

not true in the case of this package. we're just acquiring a token. let's keep it this way and improve the samples once we release GA. (similarly to https://github.com/Azure-Samples/communication-services-javascript-quickstarts/pull/73#discussion_r921096545)

File: ManageTeamsIdentityMobileAndDesktop/README.md [hunk 5]
```diff
@@ -0,0 +1,50 @@
...
+
+This sample application utilizes the [Microsoft.Identity.Client](https://docs.microsoft.com/dotnet/api/microsoft.identity.client?view=azure-dotnet) package for authentication against the Azure AD and acquisition of a token with delegated permissions. The token exchange itself is facilitated by the `Azure.Communication.Identity` package.
+
+The initialization of a Communication credential object that can be used for Calling is achieved by the `Azure.Communication.Common` package.
```
Comment (petrsvihlik) [inline ManageTeamsIdentityMobileAndDesktop/README.md:16]:
```suggestion
To be able to use the token for Calling, use it to initialize the `Azure.Communication.CommunicationTokenCredential` from the `Azure.Communication.Common` package.
```
the same problem as above

File: ManageTeamsIdentityMobileAndDesktop/README.md [hunk 6]
```diff
@@ -0,0 +1,50 @@
...
+- **./ManageTeamsIdentityMobileAndDesktop/ManageTeamsIdentityMobileAndDesktop.csproj:** Project configuration file.
+
+## Before running sample code
+
+1. Complete the [Administrator actions](https://docs.microsoft.com/azure/communication-services/quickstarts/manage-teams-identity?pivots=programming-language-csharp) from the [Manage access tokens for Teams users quickstart](https://docs.microsoft.com/azure/communication-services/quickstarts/manage-teams-identity).
+   - Take a not of Fabrikam's Azure AD Tenant ID and Contoso's Azure AD App Client ID. You'll need the values in the following steps.
+1. On the Authentication pane of your Azure AD App, add a new platform of the mobile and desktop application type with the Redirect URI of `http://localhost`.
+1. Open an instance of PowerShell, Windows Terminal, Command Prompt or equivalent and navigate to the directory that you'd like to clone the sample to.
+1. `git clone https://github.com/Azure-Samples/Communication-Services-dotnet-quickstarts.git`
+1. With the Communication Services procured in pre-requisites, add connection string,an AAD client ID and tenant ID to environment variable using below commands:
```
Comment (petrsvihlik) [inline ManageTeamsIdentityMobileAndDesktop/README.md:39]:
```suggestion
1. With the Communication Services procured in pre-requisites, add connection string, an Azure AD client ID and tenant ID to environment variable using below commands:
```

**Review by petrsvihlik — CHANGES_REQUESTED** [review]

File: ManageTeamsIdentityMobileAndDesktop/README.md [hunk 7]
```diff
@@ -0,0 +1,50 @@
...
+setx COMMUNICATION_SERVICES_CONNECTION_STRING <COMMUNICATION_SERVICES_CONNECTION_STRING>
+setx AAD_CLIENT_ID <CONTOSO_AZURE_AD_CLIENT_ID>
+setx AAD_TENANT_ID <FABRIKAM_AZURE_AD_TENANT_ID>
+
+## Run Locally
+
+1. Open `ManageTeamsIdentityMobileAndDesktop.csproj`
+2. Run the `ManageTeamsIdentityMobileAndDesktop` project
+
+You should be navigated from your browser to a standard OAuth flow with Microsoft Authentication Library (MSAL). If authentication is successful, the application receives an Azure AD access token and will be redirected to `http://localhost` where the Azure AD access token will be exchanged for a Communication access token.
```
Comment (petrsvihlik) [inline ManageTeamsIdentityMobileAndDesktop/README.md:50]:
```suggestion
You should be presented with a browser window and navigated to the Azure AD login form. If the authentication is successful, the application receives an Azure AD access token and exchanges it for a Communication access token.
```

File: ManageTeamsIdentityMobileAndDesktop/README.md (hunk 4 above)
Comment (AikoBB) [reply ManageTeamsIdentityMobileAndDesktop/README.md:12]: ok, for all platforms I will create one work item for the doc & samples improvement

File: ManageTeamsIdentityMobileAndDesktop/ManageTeamsIdentityMobileAndDesktop.csproj [hunk 8]
```diff
@@ -0,0 +1,15 @@
...
+
+  <PropertyGroup>
+    <OutputType>Exe</OutputType>
+    <TargetFramework>net6.0</TargetFramework>
+    <ImplicitUsings>enable</ImplicitUsings>
+    <Nullable>disable</Nullable>
+  </PropertyGroup>
+
+  <ItemGroup>
+    <PackageReference Include="Azure.Communication.Identity" Version="1.1.0-alpha.20220715.1" />
```
Comment (petrsvihlik) [inline ManageTeamsIdentityMobileAndDesktop/ManageTeamsIdentityMobileAndDesktop.csproj:11]: I see you changed it to the alpha version...was that intentional?

**Review by petrsvihlik — APPROVED** [review]

File: ManageTeamsIdentityMobileAndDesktop/ManageTeamsIdentityMobileAndDesktop.csproj (hunk 8 above)
Comment (petrsvihlik) [reply ManageTeamsIdentityMobileAndDesktop/ManageTeamsIdentityMobileAndDesktop.csproj]: hehe...thought so

**Review by ddematheu2 — APPROVED** [review]

**Pull request merged**